x-pack/plugins/reporting/export_types/printable_pdf/server/execute_job/index.test.js

```diff
@@ -50,8 +45,12 @@ beforeEach(() => {
       'server.basePath': ''
     }[key];
   });
+
+  generatePdfObservableFactory.mockReturnValue(jest.fn());
 });
 
+afterEach(() => generatePdfObservableFactory.mockReset());
```

Comment:
__note:__ we need this to make sure that every test tests what it should. E.g. `omits blacklisted headers` should fail if we add some header to `blacklistedHeaders` that's not blacklisted in real code, but it passed before this change since `generatePdfObservable` mock call history was shared between tests.